src/psyclone/tests/psyGen_test.py

```diff
@@ -2219,6 +2219,14 @@ def test_walk():
     binary_op_list = invoke.schedule.walk(BinaryOperation)
     assert len(binary_op_list) == 3
 
+    # Test depth restriction
+    depth_12_list = invoke.schedule.walk(BinaryOperation, depth=12)
+    assert len(depth_12_list) == 1
```

Comment:
If possible, when you move and re-write this test, could you make the check a bit more careful to ensure that the node that has been returned is definitely the one you expect and that it has the right depth (especially if my suggestion of implementing an efficiency saving comes to fruition).

Reply:
Hopefully the updated version in 53d88af1aecdb71f3734bdda20deba31cedcdea6 is better suited.

Reply:
[Now https://github.com/stfc/PSyclone/pull/2377/commits/c1a4126f33df226a4559f8d762e3f0f814161f67]